### PR TITLE
fix: dynamic k3s configuration for control planes

### PR DESCRIPTION
**Pull request by Silvest89**
Thanks to @mraerino for the original implementation I just did the error handling

Original PR
https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/881#issue-1795512839

### CODE REVIEW

File: control_planes.tf [hunk 1]
```diff
@@ -144,7 +164,7 @@ resource "null_resource" "control_planes" {
     inline = [
       "systemctl start k3s 2> /dev/null",
       <<-EOT
-      timeout 360 bash <<EOF
+      timeout 120 bash <<EOF
```
Comment (mysticaltech) [inline control_planes.tf:167]: Sometimes it takes time to start, especially on very weak nodes, so best to keep the original timeout, it was tried and tested! (Was 120 before, but that turned out not robust enough).

**Comment by mysticaltech** [comment]
@Silvest89 This is exactly what was missing, thanks! You've tested it right?

File: control_planes.tf (hunk 1 above)
Comment (Silvest89) [reply control_planes.tf:167]: Alright will fix it later.

**Comment by Silvest89** [comment]
@mysticaltech 
Updated the PR.
And yes it's tested. Spinned up a single node cluster to test some configs changes. And an invalid config ;]

image

I will open a PR for the workers after this is merged. And you can expect more PRs to support more config changes ;]

File: control_planes.tf (hunk 1 above)
Comment (mysticaltech) [reply control_planes.tf]: Ah ok, no worries then! Will fix after merging.

**Comment by mysticaltech** [comment]
@Silvest89 Love it! Merging ASAP. I needed the help, thanks for this.

**Pull request merged**